tasks/release/ReleaseVersionController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace MailPoetTasks\Release;
+
+require_once __DIR__ . '/Jira.php';
+
+class ReleaseVersionController {
+
+  /** @var Jira */
+  private $jira;
+
+  /** @var string */
+  private $project;
+
+  function __construct(Jira $jira, $project) {
+    $this->jira = $jira;
+    $this->project = $project;
+  }
+
+  static function createWithJiraCredentials($token, $user, $project) {
+    return new self(new Jira($token, $user, $project), $project);
+  }
+
+  function assignVersionToCompletedTickets($version) {
+    $output = [];
+    $output[] = "Checking version $version in $this->project";
+
+    if (!$this->checkVersion($version)) {
+      $output[] = "The version is invalid or already released";
+      return join("\n", $output);
+    }
+
+    $output[] = "Setting version $version to completed tickets in $this->project...";
+    $issues = $this->getDoneIssuesWithoutVersion();
+    $result = array_map(function ($issue) use ($version) {
+      return $this->setIssueFixVersion($issue['key'], $version);
+    }, $issues);
+    $output[] = "Done, issues processed: " . count($result);
+
+    return join("\n", $output);
+  }
+
+  function getDoneIssuesWithoutVersion() {
```

Comment:
I think this and following two methods should be private since there is no need to use them out of this class. I would keep the public interface of the class minimal.

Reply:
Fixed.